### PR TITLE
chore: Convert from `JacksonFactory` to `GsonFactory`

### LINKED ISSUE #819: Chore: Use `GsonFactory` as `JsonFactory` implementation
For a variety of reasons, mostly to do with historical security issues, the monorepo and Google security prefers Gson to Jackson, and thus `GsonFactory` to `JacksonFactory`. This change has already been made in the monorepo.

### PR DESCRIPTION
**Pull request by Capstan**
Fixes #819.

### CODE REVIEW

**Review by suztomo — APPROVED** [review]

**Comment by Capstan** [comment]
Is there anything more to do before this can be merged?

**Comment by lqiu96** [comment]
Going to re-run the post processor. Should be good to go once that check has passed.

**Comment by lqiu96** [comment]
@Capstan Checks passed. Merging.

**Pull request merged**
